source/server/admin/config_dump_handler.cc

```diff
@@ -0,0 +1,308 @@
+    // indexing.
+    // TODO(htuch): should add support for DynamicListener for multiple Any
+    // fields in the future, see
+    // https://github.com/envoyproxy/envoy/issues/9669.
+    if (field != nullptr && field->message_type() != nullptr && !field->is_repeated() &&
+        field->message_type()->full_name() == "google.protobuf.Any") {
+      if (any_field_name.empty()) {
+        any_field_name = frags[0];
+      } else {
+        // This should be structurally true due to the ConfigDump proto
+        // definition (but not for DynamicListener today).
+        ASSERT(any_field_name == frags[0],
+               "Only a single Any field in a config dump resource is supported.");
+      }
+      outer_field_mask.add_paths(frags[0]);
+      frags.erase(frags.begin());
+      inner_field_mask.add_paths(absl::StrJoin(frags, "."));
+    } else {
+      outer_field_mask.add_paths(path);
+    }
+  }
+
+  if (!any_field_name.empty()) {
+    const Protobuf::FieldDescriptor* any_field = descriptor->FindFieldByName(any_field_name);
+    if (reflection->HasField(message, any_field)) {
+      ASSERT(any_field != nullptr);
+      // Unpack to a DynamicMessage.
+      ProtobufWkt::Any any_message;
+      any_message.MergeFrom(reflection->GetMessage(message, any_field));
+      Protobuf::DynamicMessageFactory dmf;
+      const absl::string_view inner_type_name =
+          TypeUtil::typeUrlToDescriptorFullName(any_message.type_url());
+      const Protobuf::Descriptor* inner_descriptor =
+          Protobuf::DescriptorPool::generated_pool()->FindMessageTypeByName(
+              static_cast<std::string>(inner_type_name));
+      ASSERT(inner_descriptor != nullptr);
+      std::unique_ptr<Protobuf::Message> inner_message;
+      inner_message.reset(dmf.GetPrototype(inner_descriptor)->New());
+      MessageUtil::unpackTo(any_message, *inner_message);
+      // Trim message.
+      ProtobufUtil::FieldMaskUtil::TrimMessage(inner_field_mask, inner_message.get());
+      // Pack it back into the Any resource.
+      any_message.PackFrom(*inner_message);
+      reflection->MutableMessage(&message, any_field)->CopyFrom(any_message);
+    }
+  }
+  ProtobufUtil::FieldMaskUtil::TrimMessage(outer_field_mask, &message);
+}
+
+// Helper method to get the resource parameter.
+absl::optional<std::string> resourceParam(const Http::Utility::QueryParams& params) {
+  return Utility::queryParam(params, "resource");
+}
+
+// Helper method to get the mask parameter.
+absl::optional<std::string> maskParam(const Http::Utility::QueryParams& params) {
+  return Utility::queryParam(params, "mask");
+}
+
+// Helper method to get the eds parameter.
+bool shouldIncludeEdsInDump(const Http::Utility::QueryParams& params) {
+  return Utility::queryParam(params, "include_eds") != absl::nullopt;
+}
+
+} // namespace
+
+ConfigDumpHandler::ConfigDumpHandler(ConfigTracker& configTracker, Server::Instance& server)
```

Comment:
This `configTracker` should be `config_tracker` -- that's what clang-tidy is complaining about.